ion-schema/src/model/constraints/any_constraint.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+use crate::model::constraints::*;
+
+macro_rules! any_of_these {
+    (#[$derives:meta] enum $super_:ident { $($name:ident,)+ }) => {
+        #[$derives]
```

Comment:
Cool!

Reply:
I had an earlier version that would take an enum like this...

```rust
enum Foo {
    Bar { a: u8, b: bool },
    Baz { c: String, d: Vec<Element> },
}
```
...and generate structs for `Bar` and `Baz` with all of the relevant `derive`s and `getters`. It was really great until I realized that it was returning `&String` and `&Vec<_>` for things, and I'd get more idiomatic APIs if I returned `&str` and `&[_]` instead.